test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/staking/StakingSuite.java

```diff
@@ -646,6 +647,65 @@ private HapiSpec rewardPaymentsNotRepeatedInSamePeriod() {
                                 .hasPaidStakingRewards(List.of()));
     }
 
+    private HapiSpec rewardsWorkAsExpectedWithReceiverSigRequired() {
+        final long expectedTotalStakedRewardStart = ONE_HUNDRED_HBARS + ONE_HBAR;
+        final long expectedRewardRate = Math.max(0, Math.min(10 * ONE_HBAR, SOME_REWARD_RATE));
+        final long expectedRewardSumHistory =
+                expectedRewardRate / (expectedTotalStakedRewardStart / TINY_PARTS_PER_WHOLE);
+        // should be 9900990L
+        final long expectedPendingRewards = 1798608L; // base on current reward rate for 24 hours
+
+        return defaultHapiSpec("rewardsWorkAsExpectedWithReceiverSigRequired")
+                .given(
+                        overriding(STAKING_START_THRESHOLD, "" + 10 * ONE_HBAR),
+                        overriding(STAKING_REWARD_RATE, "" + SOME_REWARD_RATE),
+                        getAccountInfo(STAKING_REWARD).logged(),
+                        cryptoTransfer(tinyBarsFromTo(GENESIS, STAKING_REWARD, ONE_HBAR)))
+                .when(
+                        cryptoCreate(ALICE)
+                                .stakedNodeId(0)
+                                .balance(ONE_HUNDRED_HBARS)
+                                .receiverSigRequired(true),
+                        cryptoCreate(BOB).balance(ONE_HUNDRED_HBARS),
+                        sleepFor(INTER_PERIOD_SLEEP_MS))
```

Comment:
At some point, we need to figure out how to do this in an event-driven approach so we don't slow down testing.

Reply:
Yes, agreed, Will modify this when moving all staking tests to CI in next PR.